### PR TITLE
feat(drm_exoplayer): add drm support

### PR DESCRIPTION
**Pull request by AmeliaPessoa**

### CODE REVIEW

**Comment by AmeliaPessoa** [comment]
[(Multi-DRM) Suporte a multiplos audios e legendas no Exoplayer para vídeos DASH](https://trello.com/c/ckUP1XvG/220-multi-drm-suporte-a-multiplos-audios-e-legendas-no-exoplayer-para-v%C3%ADdeos-dash)

File: clappr/src/main/kotlin/io/clappr/player/components/Playback.kt [hunk 1]
```diff
@@ -50,35 +51,50 @@ abstract class Playback(var source: String, var mimeType: String? = null, val op
...
-        mediaOptionList.add(media)
+    open fun stop(): Boolean {
+        return false
+    }
+
+    open fun seek(seconds: Int): Boolean {
+        return false
+    }
+
+    private var mediaOptionList: LinkedList<MediaOption> = LinkedList()
```
Comment (brunovpl) [inline clappr/src/main/kotlin/io/clappr/player/components/Playback.kt]: Isn't better just use `private var mediaOptionList = LinkedList<MediaOption>()`

File: clappr/src/main/kotlin/io/clappr/player/components/Playback.kt [hunk 2]
```diff
@@ -50,35 +51,50 @@ abstract class Playback(var source: String, var mimeType: String? = null, val op
...
+    private var selectedMediaOptionList: MutableList<MediaOption> = ArrayList()
```
Comment (brunovpl) [inline clappr/src/main/kotlin/io/clappr/player/components/Playback.kt]: Isn't better just use `private var selectedMediaOptionList = ArrayList<MediaOption>()`

File: clappr/src/main/kotlin/io/clappr/player/components/Playback.kt [hunk 3]
```diff
@@ -50,35 +51,50 @@ abstract class Playback(var source: String, var mimeType: String? = null, val op
...
+    fun addAvailableMediaOption(media: MediaOption, index: Int = mediaOptionList.size) {
+        mediaOptionList.add(index, media)
     }
 
     fun availableMediaOptions(type: MediaOptionType): List<MediaOption> {
-        return mediaOptionList.filter{it.type==type}
+        return mediaOptionList.filter { it.type == type }
+    }
+
+    fun hasMediaOptionAvailable(mediaOption: MediaOption): Boolean {
```
Comment (brunovpl) [inline clappr/src/main/kotlin/io/clappr/player/components/Playback.kt]: This function is never used. Could we remove it?

File: clappr/src/main/kotlin/io/clappr/player/playback/ExoPlayerPlayBack.kt [hunk 4]
```diff
@@ -153,26 +173,47 @@ open class ExoPlayerPlayback(source: String, mimeType: String? = null, options:
...
+        player?.prepare(mediaSource)
+    }
+
+    private fun setUpRendererFactory(): DefaultRenderersFactory {
+        val rendererFactory = DefaultRenderersFactory(context,
+                buildDrmSessionManager(), DefaultRenderersFactory.EXTENSION_RENDERER_MODE_PREFER)
+        return rendererFactory
+    }
+
+    @RequiresApi(Build.VERSION_CODES.JELLY_BEAN_MR2)
```
Comment (brunovpl) [inline clappr/src/main/kotlin/io/clappr/player/playback/ExoPlayerPlayBack.kt]: As we have the sdk version check below. Isn't better remove this `@RequiresApi?`

File: clappr/src/main/kotlin/io/clappr/player/components/Playback.kt (hunk 1 above)
Comment (UelitonFreitas) [inline clappr/src/main/kotlin/io/clappr/player/components/Playback.kt]: yeap!

File: clappr/src/main/kotlin/io/clappr/player/components/Playback.kt (hunk 2 above)
Comment (UelitonFreitas) [inline clappr/src/main/kotlin/io/clappr/player/components/Playback.kt]: yes!

**Pull request merged**